UI/js/factories/factories.js

```diff
@@ -0,0 +1,92 @@
+'use strict';
+/* jshint  strict: true*/
+/* global $, errorHandler, _, dashboardApp */
+
+/**
+ * Singleton that does the requests for the courses
+ * Inner function uses the URL passed to it
+ */
+
+dashboardApp.factory('Courses', function ($http) {
+  return {
+    getCourses: function (url) {
+
+      return $http.get(url, {cache: true}).then(
+        function success(result) {
```

Comment:
This is an internal Angular cache and relates to the permanence of the model within a single page load. Say you get data for term X, then get data for term Y. The view will change to show term X and then term Y. If you then get term X again there is no request of the data from the server. If you reload the page then you have to get the data for both terms again.